kernel/kernel-api/src/main/java/io/delta/kernel/Table.java

```diff
@@ -114,6 +114,21 @@ Snapshot getSnapshotAsOfVersion(TableClient tableClient, long versionId)
     Snapshot getSnapshotAsOfTimestamp(TableClient tableClient, long millisSinceEpochUTC)
         throws TableNotFoundException;
 
+    /**
+     * Create a {@link TransactionBuilder} which can create a {@link Transaction} object to mutate
+     * the table.
+     *
+     * @param tableClient {@link TableClient} instance to use.
+     * @param engineInfo  information about the engine that is making the updates.
+     * @param operation   metadata of operation that is being performed. E.g. "insert", "delete".
+     * @return {@link TransactionBuilder} instance to build the transaction.
+     * @since 3.2.0
+     */
+    TransactionBuilder createTransactionBuilder(
```

Comment:
What happened to our discussion about having a specific API for creating new tables?

Reply:
I don't think we have time to make a decision in this release. This part needs to be revisited once 3.2 is done.